src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -39,129 +40,112 @@
-   */
-  @Autowired
-  public SampleUnitDistributor(final PlatformTransactionManager transactionManager) {
+  public SampleUnitDistributor(
+      final AppConfig appConfig,
+      final SampleUnitPublisher publisher,
+      final CollectionExerciseRepository collectionExerciseRepo,
+      final EventRepository eventRepository,
+      final SampleUnitGroupRepository sampleUnitGroupRepo,
+      final SampleUnitRepository sampleUnitRepo,
+      final ActionSvcClient actionSvcClient,
+      final PartySvcClient partySvcClient,
+      final @Qualifier("collectionExercise") StateTransitionManager<
+                  CollectionExerciseState, CollectionExerciseEvent>
+              collectionExerciseTransitionState,
+      final @Qualifier("sampleUnitGroup") StateTransitionManager<
+                  SampleUnitGroupState, SampleUnitGroupEvent>
+              sampleUnitGroupState,
+      final @Qualifier("distribution") DistributedListManager<Integer>
+              sampleDistributionListManager,
+      final PlatformTransactionManager transactionManager) {
+    this.appConfig = appConfig;
+    this.publisher = publisher;
+    this.collectionExerciseRepo = collectionExerciseRepo;
+    this.eventRepository = eventRepository;
+    this.sampleUnitGroupRepo = sampleUnitGroupRepo;
+    this.sampleUnitRepo = sampleUnitRepo;
+    this.actionSvcClient = actionSvcClient;
+    this.partySvcClient = partySvcClient;
+    this.collectionExerciseTransitionState = collectionExerciseTransitionState;
+    this.sampleUnitGroupState = sampleUnitGroupState;
+    this.sampleDistributionListManager = sampleDistributionListManager;
     this.transactionTemplate = new TransactionTemplate(transactionManager);
     this.transactionTemplate.setTimeout(TRANSACTION_TIMEOUT);
   }
 
   /**
-   * Distribute SampleUnits for a CollectionExercise.
+   * Distribute SampleUnits for a CollectionExercise
    *
-   * @param exercise for which to distribute sample units.
+   * @param exercise for which to distribute sample units
    */
   public void distributeSampleUnits(CollectionExercise exercise) {
 
+    List<ExerciseSampleUnitGroup> sampleUnitGroups = new ArrayList<>();
     try {
-      List<ExerciseSampleUnitGroup> sampleUnitGroups = retrieveSampleUnitGroups(exercise);
-
-      for (ExerciseSampleUnitGroup anExerciseSampleUnitGroup : sampleUnitGroups) {
-        distributeSampleUnits(exercise, anExerciseSampleUnitGroup);
-      }
-
-      if (!sampleUnitGroups.isEmpty()) {
-        collectionExerciseTransitionState(exercise);
-      }
-
-    } catch (LockingException | CTPException ex) {
-      log.error("Distribution failed", ex);
-    } finally {
-      try {
-        sampleDistributionListManager.deleteList(DISTRIBUTION_LIST_ID, true);
-      } catch (LockingException ex) {
-        log.error("Failed to release sampleDistributionListManager data", ex);
-      }
+      sampleUnitGroups = retrieveSampleUnitGroups(exercise);
+    } catch (LockingException ex) {
```

Comment:
this seems very arbitrary. It's not very clear what would cause a locking exception when nothing in this function refers to locking